### ISSUE TITLE
NPE on JSON deserialization

### ISSUE THREAD

**Issue by rock3r** [frakbot/FWeather#2]
When the JSON is malformed, there could be a `NullPointerException`:

```
java.lang.NullPointerException
at org.json.JSONTokener.nextCleanInternal(JSONTokener.java:116)
at org.json.JSONTokener.nextValue(JSONTokener.java:94)
at org.json.JSONObject.<init>(JSONObject.java:154)
at org.json.JSONObject.<init>(JSONObject.java:171)
at net.frakbot.FWeather.JSONWeatherParser.getWeather(JSONWeatherParser.java:42)
at net.frakbot.FWeather.UpdaterService.getWeather(UpdaterService.java:348)
at net.frakbot.FWeather.UpdaterService.onHandleIntent(UpdaterService.java:70)
at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:213)
at android.os.HandlerThread.run(HandlerThread.java:60)
```

This is a really rare condition, but should be fixed anyway.

**Comment by frapontillo** [comment]
Does this mean that OpenWeather gives us a wrongly serialized object? Have you been able to catch an example file?

**Comment by rock3r** [comment]
No, I was on the go. I suspect there was some problem downloading, or maybe
some momentary down time. Should just catch and retry a couple of times, in
my opinion

**Comment by rock3r** [comment]
Just happened again. I turned off the screen while on the configuration activity, then turned it on again.

```
07-06 02:33:55.180 E/FWeatherHttp(20421): Error while fetching the weather
07-06 02:33:55.180 E/FWeatherHttp(20421): java.io.EOFException
07-06 02:33:55.180 E/FWeatherHttp(20421):   at libcore.io.Streams.readAsciiLine(Streams.java:203)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at libcore.net.http.HttpEngine.readResponseHeaders(HttpEngine.java:573)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at libcore.net.http.HttpEngine.readResponse(HttpEngine.java:821)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at libcore.net.http.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:283)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at libcore.net.http.HttpURLConnectionImpl.getInputStream(HttpURLConnectionImpl.java:177)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at net.frakbot.FWeather.updater.weather.WeatherHttpClient.getCityWeatherJsonData(WeatherHttpClient.java:66)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at net.frakbot.FWeather.updater.UpdaterService.getWeather(UpdaterService.java:378)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at net.frakbot.FWeather.updater.UpdaterService.onHandleIntent(UpdaterService.java:108)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at android.os.Handler.dispatchMessage(Handler.java:99)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at android.os.Looper.loop(Looper.java:137)
07-06 02:33:55.180 E/FWeatherHttp(20421):   at android.os.HandlerThread.run(HandlerThread.java:60)
07-06 02:33:55.180 W/dalvikvm(20421): threadid=12: thread exiting with uncaught exception (group=0x40a79930)
07-06 02:33:55.180 E/AndroidRuntime(20421): FATAL EXCEPTION: IntentService[UpdaterService]
07-06 02:33:55.180 E/AndroidRuntime(20421): java.lang.NullPointerException
07-06 02:33:55.180 E/AndroidRuntime(20421):     at org.json.JSONTokener.nextCleanInternal(JSONTokener.java:116)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at org.json.JSONTokener.nextValue(JSONTokener.java:94)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at org.json.JSONObject.<init>(JSONObject.java:154)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at org.json.JSONObject.<init>(JSONObject.java:171)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at net.frakbot.FWeather.updater.weather.JSONWeatherParser.getWeather(JSONWeatherParser.java:42)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at net.frakbot.FWeather.updater.UpdaterService.getWeather(UpdaterService.java:386)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at net.frakbot.FWeather.updater.UpdaterService.onHandleIntent(UpdaterService.java:108)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at android.os.Handler.dispatchMessage(Handler.java:99)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at android.os.Looper.loop(Looper.java:137)
07-06 02:33:55.180 E/AndroidRuntime(20421):     at android.os.HandlerThread.run(HandlerThread.java:60)
07-06 02:34:00.595 W/InputMethodManagerService(536): Got RemoteException sending setActive(false) notification to pid 20421 uid 10110
```

**Comment by rock3r** [comment]
Fixed as of ceab4ee
